macrosynergy/dataquery/auth.py

```diff
@@ -98,32 +108,41 @@ def __init__(
         self.key: str = self.valid_path(key, "key")
         self.crt: str = self.valid_path(crt, "crt")
 
-        # For debugging purposes save last request response
+        # For debugging purposes save last request response.
         self.status_code: Optional[int] = None
         self.last_response: Optional[str] = None
         self.last_url: Optional[str] = None
 
     @staticmethod
     def valid_path(directory: str, file_type: str) -> Optional[str]:
-        """Validates the key & certificate exist in the referenced directory."""
-        assert isinstance(directory, str), (
-            f"{file_type:s} file must be a <str> not {directory}{type(directory)})."
-        )
+        """Validates the key & certificate exist in the referenced directory.
 
-        if not os.path.exists(directory) and os.path.isfile(directory):
-            # TODO should this not raise an error (exception)?
-            print(f"Not a valid file path on the system: {directory} - return None")
-            return None
+        :param <str> directory: directory hosting the respective files.
+        :param <str> file_type: parameter used to distinguish between the certificate or
+            key being received.
+
+        """
+        dir_error = f"{file_type:s} file must be a <str> not <{type(directory)}>."
+        assert isinstance(directory, str), dir_error
 
-        return directory
+        condition = (os.path.exists(directory) and os.path.isfile(directory))
+        if condition:
+            return directory
+        else:
+            print(f"The directory received, {directory}, does not contain the respective"
```

Comment:
This works, but why do we not just raise a `OSError` if the condition is not meet:

```python
if not (os.path.exists(directory) and os.path.isfile(directory)):
    raise OSError(f"The directory received, {directory}, does not contain the respective")
```
That way we know exactly where and what has failed.

Reply:
Change made.